probe-rs-tools/src/bin/probe-rs/util/common_options.rs

```diff
@@ -415,86 +415,90 @@ CARGO BUILD OPTIONS:
 pub enum OperationError {
     #[error("No connected probes were found.")]
     NoProbesFound,
+
     #[error("Failed to open the ELF file '{path}' for flashing.")]
     #[allow(dead_code)]
     FailedToOpenElf {
         #[source]
         source: std::io::Error,
         path: PathBuf,
     },
+
     #[error("Failed to load the ELF data.")]
     #[allow(dead_code)]
     FailedToLoadElfData(#[source] FileDownloadError),
+
     #[error("Failed to open the debug probe.")]
     FailedToOpenProbe(#[source] DebugProbeError),
+
     #[error("{} probes were found: {}", .list.len(), print_list(.list))]
     MultipleProbesFound { list: Vec<DebugProbeInfo> },
+
     #[error("The flashing procedure failed for '{path}'.")]
     FlashingFailed {
-        #[source]
         source: FlashError,
         target: Box<Target>, // Box to reduce enum size
         target_spec: Option<String>,
         path: PathBuf,
     },
+
     #[error("Failed to open the chip description '{path}'.")]
     ChipDescriptionNotFound {
-        #[source]
         source: std::io::Error,
         path: PathBuf,
     },
+
     #[error("Failed to parse the chip description '{path}'.")]
     FailedChipDescriptionParsing {
-        #[source]
         source: RegistryError,
         path: PathBuf,
     },
+
     #[error("Failed to change the working directory to '{path}'.")]
     FailedToChangeWorkingDirectory {
-        #[source]
         source: std::io::Error,
         path: PathBuf,
     },
+
     #[error("Failed to build the cargo project at '{path}'.")]
     FailedToBuildExternalCargoProject {
-        #[source]
         source: ArtifactError,
         path: PathBuf,
     },
+
     #[error("Failed to build the cargo project.")]
     FailedToBuildCargoProject(#[source] ArtifactError),
+
     #[error("The chip '{name}' was not found in the database.")]
-    ChipNotFound {
-        #[source]
-        source: RegistryError,
-        name: String,
-    },
+    ChipNotFound { source: RegistryError, name: String },
+
     #[error("The protocol '{protocol}' could not be selected.")]
     FailedToSelectProtocol {
-        #[source]
         source: DebugProbeError,
         protocol: WireProtocol,
     },
+
     #[error("The protocol speed could not be set to '{speed}' kHz.")]
-    FailedToSelectProtocolSpeed {
-        #[source]
```

Comment:
Removing source is not correct.

Reply:
It is, the field name implicitly makes it a `#[source]`

Reply:
What @bugadani said